base/operators.jl

```diff
@@ -1035,11 +1035,20 @@ end
 ∘(f, g, h...) = ∘(f ∘ g, h...)
 
 function show(io::IO, c::ComposedFunction)
-    show(io, c.outer)
+    if c.outer |> Symbol |> Base.isoperator
+        print(io, '(', c.outer, ')')
+    else
+        show(io, c.outer)
+    end
```

Comment:
Probably cleaner to have a
```jl
showfunction(io::IO, f) = show(io, x)
showfunction(io::IO, f::Function) = Base.isoperator(Symbol(f)) ? print(io, '(', f, ')') : show(io, f)
showfunction(io::IO, f::ComposedFunction) = print(io, '(', f, ')')
```
and then call this for both `inner` and `outer`.

Reply:
Note that `print` output` can differ from `show`, so we want to be careful to distinguish the two:
```jl
julia> show(sin ∘ cos)
sin ∘ cos
julia> print(sin ∘ cos)
ComposedFunction{typeof(sin), typeof(cos)}(sin, cos)
```

Reply:
We might also want to define:
```jl
function show_function(io::IO, c::ComposedFunction, compact::Bool)
    if compact
        show(io, f)
    else
        print(io, "ComposedFunction(")
        show_function(io, c.outer, false)
        print(io, ", ")
        show_function(io, c.inner, false)
        print(io, ')')
    end
end
````
with a fallback
```jl
show_function(io::IO, x, compact::Bool) = show(io, x)
```
in case the `inner` or `outer` field is not a `Function`.

This way, `print` won't print the `typeof` (which is redundant), and the `compact` parameter will nest.

Reply:
This behavior of `print` looks like a bug to me; if `print` is called then `show_function` should fall back to `show` instead of `show_default`. So I don't think methods should be added to `show_function`, we can fix it instead.